base_local_planner/src/speed_limiters/shadow_speed_limiter.cpp

```diff
@@ -92,22 +96,51 @@ bool ShadowSpeedLimiter::calculateLimits(double& max_allowed_linear_vel, double&
   map_grid_.resetPathDist();
   map_grid_.setUnadjustedGoal(*(costmap_->getCostmap()), pose);
 
+  double points_inside = 0;
+  double points_total = 0;
+  double max_vel = max_linear_velocity_;
   // Find nearest object via brushfire distance
+
+  double max_cost = map_grid_.obstacleCosts();
   for (const auto& obj : (*objects))
+  {	
+	// Get the brushfire distance to the obstacle
+   	double distance = getMapGridDistance(obj)/costmap_->getCostmap()->getResolution();
+	if (fabs(getAngle(pose_no_offset, obj)) < params_.half_angle && distance < max_cost)
+	{	
+    	// Convert it to a velocity
+		double distance = getMapGridDistance(obj);
+    	double velocity = distanceToVelocity(distance);
+		max_vel = std::min(velocity, max_vel);
+
+		points_inside++;
+	}
+	points_total++;
+  }
+  if (points_inside / points_total > params_.shadow_threshold)
```

Comment:
Is this disabled by default or enabled by default?

Reply:
Feature is turned off for now by setting shadow_threshold to 0.